llama_index/logger/__init__.py

```diff
@@ -1,5 +1,14 @@
 """Init params."""
 
+import logging
+from logging import NullHandler
+
 from llama_index.logger.base import LlamaLogger
 
 __all__ = ["LlamaLogger"]
+
+logger = logging.getLogger(__name__)
```

Comment:
By initialized the logger here, is the logger on module level still getting information about the module where it's executed?

I think it's pretty common to name your logger based on module/class, that way we can easily identify from which module/class the logs come from.

An example from java
```
logger = Logger.getLogger(MyClass.class.getName())
```

On python context

> __name__ is one such special variable. If the source file is executed as the main program, the interpreter sets the __name__ variable to have a value “__main__”. If this file is being imported from another module, __name__ will be set to the module’s name.

Reply:
- The logger name will be standardized across the entire library. In it's current state, the logger is named `llama_index.logger`. As you point out, the `__name__` dunder resolves to the file name.

- Python's `logging` framework supports [Formatters](https://docs.python.org/3/library/logging.html#formatter-objects), which includes function name, line number, etc in context. So a developer can configure the context they want.

Reply:
imo I think it would make sense to at least have some opinionated initial logger format to capture that info

Reply:
Agree, the default formatter is preferred. I think one of the principles of LlamaIndex is, that you can start as simple as possible with minimum configuration, but you have the option to override almost everything.
By providing default formatted, the new user can benefit from that without knowing too much, and the advanced user can override the formatter as they need.

I think we can follow SqlAlchemy implementation of that.
https://github.com/sqlalchemy/sqlalchemy/blob/main/lib/sqlalchemy/log.py#L53